src/libstore/realisation.hh

```diff
@@ -28,8 +39,60 @@ struct Realisation {
 
     nlohmann::json toJSON() const;
     static Realisation fromJSON(const nlohmann::json& json, const std::string& whence);
+
+    StorePath getPath() const { return outPath; }
+
+    GENERATE_CMP(Realisation, me->id, me->outPath);
+};
+
+struct OpaquePath {
+    StorePath path;
+
+    StorePath getPath() const { return path; }
+
+    GENERATE_CMP(OpaquePath, me->path);
 };
 
-typedef std::map<DrvOutput, Realisation> DrvOutputs;
+
+/**
+ * A store path with all the history of how it went into the store
+ */
+struct RealisedPath {
```

Comment:
On a separate note, I think it would be good to make a plan for reducing into fewer types this, `StorePathWithOutputs`, `Buildable`, and perhaps even the separation of `inputsDrv` vs `inputSrcs`. We've had too many similar data types for too long, but only recently have we been bringing them closer together.